Fastpack/Error.ml

```diff
@@ -1,11 +1,58 @@
 module Loc = FlowParser.Loc
 module Scope = FastpackUtil.Scope
 
+type color = Cyan | Red | Black | White;;
+type font = Regular | Bold;;
+
+let print_with_color ?font:(font=Regular) str col = 
+  let col = match col with 
+    | Cyan -> "36"
+    | Red -> "31"
+    | Black -> "30"
+    | White -> "37"
+  in
+  let f = match font with 
+    | Regular -> "0"
+    | Bold -> "1"
+  in
+  "\027[" ^ f ^ ";" ^ col ^ "m" ^ str ^ "\027[0m"
+
+let get_codeframe (loc: Loc.t) lines = 
+  let startLine = max 0 (loc.start.line - 2) in
+  let endLine = min (List.length lines) (loc._end.line + 1) in
+  let codeframe = (List.filter_map (fun (i, line) -> 
+      if startLine <= i  && i <= endLine then 
+        Some (i, line) 
+      else None
+    ) lines) 
+  in
+  let maxLineNo = List.fold_left (fun n (i, _) -> max n i) 0 lines in
+  let maxDigits = String.length (string_of_int maxLineNo) in
+  let formatted = List.map (fun (i, line) -> 
+      let isErrorLine = loc.start.line <= i && i <= loc._end.line in
+      let isTTY = (Unix.isatty Unix.stderr) in
+      let lineNo = String.pad (maxDigits) (string_of_int i) in
+      match (isErrorLine, isTTY) with
+      | false, _ -> lineNo ^ " │ " ^ line 
+      | true, false -> 
+        let (offset, length) = (loc.start.column + 1), (loc._end.column - loc.start.column) in
```

Comment:
Same as above: `let offset, length = (loc.start.column + 1), (loc._end.column - loc.start.column)`. They are not needed in the right-hand side as well, but I would probably keep them just for readability.

Reply:
done